src/v2/guide/reactivity.md

```diff
@@ -20,7 +20,11 @@ Every component instance has a corresponding **watcher** instance, which records
 
 ## Change Detection Caveats
 
-Due to the limitations of modern JavaScript (and the abandonment of `Object.observe`), Vue **cannot detect property addition or deletion**. Since Vue performs the getter/setter conversion process during instance initialization, a property must be present in the `data` object in order for Vue to convert it and make it reactive. For example:
+Due to limitations in JavaScript, there are types of changes that Vue **cannot detect**. However, there are ways to circumvent them to preserve reactivity.
+
+### With Objects
```

Comment:
Is it "Change detection caveats _for_ objects" or "_with_ objects"? Not a native speaker myself, so…

Reply:
I asked a native speaker colleague and indeed, "for" is best!